crypto/bn256/gnark/g1.go

```diff
@@ -0,0 +1,51 @@
+package bn256
+
+import (
+	"math/big"
+
+	"github.com/consensys/gnark-crypto/ecc/bn254"
```

Comment:
Me not being a cryptographer, bear with me... This is package `bn256`, but it operates on `bn254` operands? Why the discrepancy?

Reply:
This was mainly poor historical naming: bn128/bn256/bn254 all refer to the same thing -- the discrepancy in numbers refer to different properties of the curve.

The 128 referred to the level of security, however it was found out that it did not have this level of security and then it was changed to bn256 or bn254 depending on who you ask